lib/encoder_impl.cc

```diff
@@ -286,6 +287,12 @@ void encoder_impl::set_ps(std::string ps) {
 
 		std::memset(PS, ' ', sizeof(PS));
 		std::memcpy(PS, ps.c_str(), len);
+
+        using std::cout;
+        using std::endl;
+        cout << "PS set to \"" << ps << "\"" << endl;
+        rebuild();
+        rebuild();
```

Comment:
for a reason. one rebuild() alone doesn't work. (don't ask me why, all this isn't better than a quick hack)